msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/MzMLSpectrum.java

```diff
@@ -181,7 +208,12 @@ public Integer getNumberOfDataPoints() {
   @Override
   public double[] getMzValues() {
     double[] mzValues = null;
-    this.mzArrayLength = getMzBinaryDataInfo().getArrayLength();
+    Integer numOfDataPoints = this.numOfDataPoints;
```

Comment:
Sorry, I'll fix it. I was initially intending to set that value to the one contained in BinaryDataInfo. Should we also throw an exception in this case? http://www.peptideatlas.org/tmp/mzML1.1.0.html#binaryDataArray